### ISSUE TITLE
Eta front-end

### ISSUE THREAD

**Issue by romac** [epfl-lara/stainless#475]
Blocked by https://twitter.com/rahulmutt/status/1113791731772743680

**Comment by vkuncak** [comment]
With @redelmann 's new parsing combinator library we should be able to have an alternative Haskhellish front end rather quickly.

**Comment by romac** [comment]
Yeah I am hoping to be able to implement https://github.com/epfl-lara/stainless/issues/494 and/or merge the WASM backend + add WASI [0] support so that we can have our own little and lightweight language, tailored to the features we support well in Stainless.

[0] https://hacks.mozilla.org/2019/03/standardizing-wasi-a-webassembly-system-interface/

**Comment by vkuncak** [comment]
Did you see Z3 compiled to WASM?

https://github.com/cpitclaudel/z3.wasm
